### PR TITLE
Add `--profile-cpu` and `--profile-mem` options

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:
Those allow to write a CPU and memory profile over CRI-O's complete
lifetime. The options write an adequate pprof memory or CPU profile to
the target files, which can be used for debugging purposes later on.

#### Which issue(s) this PR fixes:

None
#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
Added `--profile-cpu` and `--profile-mem` CLI flags to write pprof profiles to a file when CRI-O exists.
```

### CODE REVIEW

**Comment by TomSweeneyRedHat** [comment]
LGTM
assuming happy tests

File: cmd/crio/main.go [hunk 1]
```diff
@@ -347,6 +368,22 @@ func main() {
...
+		memProfilePath := c.String("profile-mem")
+		if memProfilePath != "" {
+			logrus.Infof("Creating memory profile in: %v", memProfilePath)
+
+			file, err := os.Create(memProfilePath)
+			if err != nil {
+				return errors.Wrap(err, "could not create memory profile")
+			}
+			defer file.Close()
+			runtime.GC()
```
Comment (mrunalp) [inline cmd/crio/main.go:380]: Why do we need this?

Comment (saschagrunert) [reply cmd/crio/main.go:380]: To update the statistics of the profile.

For example, let's compare those three memory profiles. What I did is starting CRI-O and running `crictl version` and `crictl images` consecutively.

1. With the http pprof endpoint: `sudo curl --unix-socket /var/run/crio/crio.sock "http://localhost/debug/pprof/heap"`
profile001

2. With the http pprof endpoint: `sudo curl --unix-socket /var/run/crio/crio.sock "http://localhost/debug/pprof/heap"` and running the GC with SIGUSR2: `sudo kill -SIGUSR2 $(pgrep crio)`
profile002

3. With the new `--profile-mem` option
profile003

Comment (mrunalp) [reply cmd/crio/main.go:380]: I meant the call to runtime.GC() :)

Comment (saschagrunert) [reply cmd/crio/main.go:380]: Oh it's part of the original example: https://pkg.go.dev/runtime/pprof

**Pull request merged**